src/formatting_stack/reporters/impl.clj

```diff
@@ -17,3 +17,19 @@
                                        (string/replace (re-pattern (str "^" (System/getProperty "user.dir")))
                                                        "")
                                        (string/replace #"^/" ""))))
+
+(speced/defn truncate-line-wise [^string? s, length]
+  (if (= s "\n")
+    s
+    (->> (string/split s #"\n")
+         (map (fn [s]
+                (let [suffix "…"
+                      string-length (count s)
+                      suffix-length (count suffix)]
+                  (if (<= string-length length)
+                    s
+                    (str (subs s
+                               0
+                               (- length suffix-length))
+                         suffix)))))
```

Comment:
I guess you could `defn` this, overflow'ing a single line sounds useful on its own.